cmake/org.eclipse.cdt.cmake.core.tests/src/org/eclipse/cdt/cmake/core/CMakeBuildConfigurationTests.java

```diff
@@ -0,0 +1,208 @@
+		IOsOverrides linuxOverrides = cMakeProperties.getLinuxOverrides();
+		CMakeGenerator linuxGenerator = linuxOverrides.getGenerator();
+		assertThat(linuxGenerator, is(CMakeGenerator.UnixMakefiles));
+	}
+
+	/**
+	 * Test for {@link ICMakeProperties#setExtraArguments()}
+	 * This is a different extraArguments to IOsOverrides#setExtraArguments().
+	 * Presumably ICMakeProperties#setExtraArguments() are platform agnostic extra arguments, where as
+	 * IOsOverrides#setExtraArguments() can be set different for Linux and Windows.
+	 */
+	@Test
+	public void getCMakePropertiesTestSetExtraArguments() throws Exception {
+		// Create a C Build Configuration using the default build config and an arbitrary name
+		CMakeBuildConfigurationExtended cmBuildConfig = new CMakeBuildConfigurationExtended(buildConfig,
+				"cmBuildConfigName", mockToolchain) {
+
+			@Override
+			public ICMakeProperties getCMakeProperties() {
+				ICMakeProperties properties = super.getCMakeProperties();
+				properties.setExtraArguments(
+						new ArrayList<>((List.of("-DplatformAgnosticArgsTest0=0", "-DplatformAgnosticArgsTest1=1"))));
+				return properties;
+			}
+		};
+		// Call the new method on ICMakeBuildConfiguration to get the default CMake properties.
+		ICMakeProperties cMakeProperties = cmBuildConfig.getCMakeProperties();
+		List<String> extraArguments = cMakeProperties.getExtraArguments();
+		assertThat(extraArguments, contains("-DplatformAgnosticArgsTest0=0", "-DplatformAgnosticArgsTest1=1"));
+	}
+
+	/**
+	 * Test for {@link IOsOverrides#setExtraArguments()}
+	 */
+	@Test
+	public void getCMakePropertiesTestIOsOverridesSetExtraArguments() throws Exception {
+		// Create a C Build Configuration using the default build config and an arbitrary name
+		CMakeBuildConfigurationExtended cmBuildConfig = new CMakeBuildConfigurationExtended(buildConfig,
+				"cmBuildConfigName", mockToolchain);
+		// Call the new method on ICMakeBuildConfiguration to get the default CMake properties.
+		ICMakeProperties cMakeProperties = cmBuildConfig.getCMakeProperties();
+
+		// Get overrides for Windows host and check the default value for getExtraArguments.
+		IOsOverrides windowsOverrides = cMakeProperties.getWindowsOverrides();
+		List<String> winExtraArguments = windowsOverrides.getExtraArguments();
+		assertThat(winExtraArguments, contains("-Dtest0=0", "-Dtest1=1"));
+
+		// Get overrides for Linux host and check the default value for getExtraArguments.
+		IOsOverrides linuxOverrides = cMakeProperties.getLinuxOverrides();
+		List<String> linuxExtraArguments = linuxOverrides.getExtraArguments();
+		assertThat(linuxExtraArguments, contains("-DLinuxtest0=0", "-DLinuxtest1=1"));
+	}
+
+	private class CMakeBuildConfigurationExtended extends CMakeBuildConfiguration {
+
+		public CMakeBuildConfigurationExtended(IBuildConfiguration config, String name, IToolChain toolChain) {
+			super(config, name, toolChain);
+		}
+
```

Comment:
A key set of tests is missing still, there are no tests for when an extender of CMakeBuildConfiguration wants to provide different defaults for things like the generator and extra args.